server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/MailboxGetMethod.scala

```diff
@@ -0,0 +1,90 @@
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.method
+
+import eu.timepit.refined.auto._
+import javax.inject.Inject
+import org.apache.james.jmap.json.Serializer
+import org.apache.james.jmap.mail._
+import org.apache.james.jmap.model.Invocation.{Arguments, MethodName}
+import org.apache.james.jmap.model.State.INSTANCE
+import org.apache.james.jmap.model.{AllMailboxesItem, Invocation, MailboxFactory}
+import org.apache.james.jmap.utils.quotas.{QuotaLoaderWithDefaultPreloaded, QuotaLoaderWithDefaultPreloadedFactory}
+import org.apache.james.mailbox.model.MailboxMetaData
+import org.apache.james.mailbox.model.search.MailboxQuery
+import org.apache.james.mailbox.{MailboxManager, MailboxSession}
+import org.reactivestreams.Publisher
+import play.api.libs.json.{JsError, JsObject, JsSuccess}
+import reactor.core.scala.publisher.{SFlux, SMono}
+import reactor.core.scheduler.Schedulers
+
+class MailboxGetMethod @Inject() (serializer: Serializer,
+                                  mailboxManager: MailboxManager,
+                                  quotaFactory : QuotaLoaderWithDefaultPreloadedFactory,
+                                  mailboxFactory: MailboxFactory) extends Method {
+  override val methodName: MethodName = MethodName("Mailbox/get")
+
+  override def process(invocation: Invocation, mailboxSession: MailboxSession): Publisher[Invocation] = {
+    asMailboxGetRequest(invocation.arguments)
+      .flatMap(mailboxGetRequest => getMailboxes(mailboxGetRequest, mailboxSession)
+        .collectSeq()
+        .map(mailboxes => MailboxGetResponse(
+          accountId = mailboxGetRequest.accountId,
+          state = INSTANCE,
+          list = mailboxes.toList,
+          notFound = NotFound(Nil)))
+        .map(mailboxGetResponse => Invocation(
+          methodName = methodName,
+          arguments = Arguments(serializer.serialize(mailboxGetResponse).as[JsObject]),
+          methodCallId = invocation.methodCallId)))
+  }
+
+  private def asMailboxGetRequest(arguments: Arguments): SMono[MailboxGetRequest] = {
+    serializer.deserializeMailboxGetRequest(arguments.value.toString) match {
+      case JsSuccess(mailboxGetRequest, _) => SMono.just(mailboxGetRequest)
+      case JsError(_) => SMono.raiseError(new IllegalArgumentException("Invalid MailboxGetRequest"))
```

Comment:
it's unfortunate to loose the valuable validation information contained in the JsError